lib/cn/kvs_mblk_desc.c

```diff
@@ -78,8 +78,13 @@ mblk_madvise(const struct kvs_mblk_desc *md, size_t off, size_t len, int advice)
     if (len == 0)
         return 0;
 
-    assert(off + len <= md->wlen_pages * PAGE_SIZE);
     assert(md->map_base);
+    assert(off < md->wlen_pages * PAGE_SIZE);
+
+    if (off + len > md->wlen_pages * PAGE_SIZE)
+        len = md->wlen_pages * PAGE_SIZE - off;
+
+    assert(off + len <= md->wlen_pages * PAGE_SIZE);
```

Comment:
this assert feels unnecessary since it exactly matches the preceding `if` condition.